R/object_name_linter.R

```diff
@@ -45,7 +45,11 @@ object_name_xpath <- local({
 #'
 #' @param styles A subset of
 #'   \Sexpr[stage=render, results=rd]{lintr:::regexes_rd}. A name should
-#'   match at least one of these styles.
+#'   match at least one of these styles. The `"symbols"` style refers to
+#'   names containing *only* non-alphanumeric characters; e.g., `%+%` from
+#'   ggplot2 or `%>%` from magrittr will not generate lint markers, whereas
+#'   `%in%` (containing both alphanumeric *and* non-alphanumeric characters)
```

Comment:
Actually this is considered to be the same as `"in"` for the purpose of `object_name_linter()`, so it will match any lowercase style.
`` lubridate::`%m+%` `` is a better example I think.

Reply:
Fair enough -- happy to make that change, but I've just found that `%>%` will not generate a lint regardless of whether `"symbols"` is included; is that expected?

```{r}
# R/test.R
library(magrittr)
iris %>%
  nrow()
```

```{r}
# Does not generate a lint
lint("R/test.R", linters = list(object_name_linter(styles = "snake_case")))
```

Reply:
The linter only lints object names that you define. Your example doesn't contain any assignments that could be linted.

Reply:
Ah right, okay. How does it look now?